### ISSUE TITLE
Syncing usage data from cloud not working - interface conversion issue

### ISSUE THREAD

**Issue by maathor** [infracost/infracost#1675]
**Context:**

We're using infracost to generate some finops simulation for each of our squads.

We're currently using :
- infracost `v0.9.24`
- terraform `v1.0.5`

On our GHA, we're :
- generating some plan, with terraform
- convert it into json with terraform.
- And based on this json, we're want to sync them with our usage on each AWS account.

we're using:
`infracost breakdown --sync-usage-file --usage-file infracost-usage-apps-beta.yml --path tfplan.json --log-level debug`

That command fails with:
```
time="2022-05-20T12:03:39+02:00" level=info msg="Detected Terraform plan JSON file at tfplan.json"
time="2022-05-20T12:03:39+02:00" level=debug msg="Specified usage file already exists, no overriding"
time="2022-05-20T12:03:39+02:00" level=info msg="Starting: Extracting only cost-related params from terraform"
time="2022-05-20T12:03:39+02:00" level=debug msg="Checking if tfplan.json is a git repo"
time="2022-05-20T12:03:40+02:00" level=debug msg="reference does not specify a count: using resource module.elastic_beanstalk_environment_app.module.elastic_beanstalk_environment_app[0].data.aws_lb.ads_application_loadbalancer[0] for for module.elastic_beanstalk_environment_app.module.elastic_beanstalk_environment_app[0].aws_route53_record.app_record[0].alias.0.name"
time="2022-05-20T12:03:40+02:00" level=debug msg="Loading usage data from Infracost provider resources"
time="2022-05-20T12:03:40+02:00" level=debug msg="Loading usage data from Infracost provider resources"
time="2022-05-20T12:03:40+02:00" level=info msg="Starting: Syncing usage data from cloud"
time="2022-05-20T12:03:40+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Invocations (region: eu-west-1, FunctionName: PlatformPublishingPostIndexer)"
time="2022-05-20T12:03:40+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Invocations (region: eu-west-1, FunctionName: PlatformPublishingCalendarEventIndexUpdater)"
time="2022-05-20T12:03:40+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Invocations (region: eu-west-1, FunctionName: PlatformPublishingCalendarEventIndexer)"
time="2022-05-20T12:03:40+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Invocations (region: eu-west-1, FunctionName: PlatformPublishingInstagramPublisher)"
time="2022-05-20T12:03:40+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Invocations (region: eu-west-1, FunctionName: PlatformPublishingBatchRunner)"
time="2022-05-20T12:03:40+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Invocations (region: eu-west-1, FunctionName: PlatformPublishingQueueMigrator)"
time="2022-05-20T12:03:40+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Invocations (region: eu-west-1, FunctionName: PlatformPublishingInboxUpdatePostUpdater)"
time="2022-05-20T12:03:40+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Invocations (region: eu-west-1, FunctionName: PlatformPublishingYoutubePublisher)"
time="2022-05-20T12:03:40+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Invocations (region: eu-west-1, FunctionName: PlatformPublishingUnapprovedQueuedPostSkipper)"
time="2022-05-20T12:03:40+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Invocations (region: eu-west-1, FunctionName: PlatformPublishingVideoConverter)"
time="2022-05-20T12:03:40+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Invocations (region: eu-west-1, FunctionName: PlatformPublishingPostIndexUpdater)"
time="2022-05-20T12:03:40+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Invocations (region: eu-west-1, FunctionName: PlatformPublishingFacebookPublisher)"
time="2022-05-20T12:03:40+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Invocations (region: eu-west-1, FunctionName: PlatformPublishingTwitterPublisher)"
time="2022-05-20T12:03:40+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Invocations (region: eu-west-1, FunctionName: PlatformPublishingThumbnailGenerator)"
time="2022-05-20T12:03:40+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Invocations (region: eu-west-1, FunctionName: PlatformPublishingQueueItemMigrator)"
time="2022-05-20T12:03:40+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Invocations (region: eu-west-1, FunctionName: PlatformPublishingGooglePublisher)"
time="2022-05-20T12:03:41+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Duration (region: eu-west-1, FunctionName: PlatformPublishingPostIndexUpdater)"
time="2022-05-20T12:03:42+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Duration (region: eu-west-1, FunctionName: PlatformPublishingBatchRunner)"
time="2022-05-20T12:03:42+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Duration (region: eu-west-1, FunctionName: PlatformPublishingInboxUpdatePostUpdater)"
time="2022-05-20T12:03:42+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Invocations (region: eu-west-1, FunctionName: PlatformPublishingLinkedinPublisher)"
time="2022-05-20T12:03:42+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Duration (region: eu-west-1, FunctionName: PlatformPublishingThumbnailGenerator)"
time="2022-05-20T12:03:42+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Duration (region: eu-west-1, FunctionName: PlatformPublishingFacebookPublisher)"
time="2022-05-20T12:03:42+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Duration (region: eu-west-1, FunctionName: PlatformPublishingCalendarEventIndexer)"
time="2022-05-20T12:03:42+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Duration (region: eu-west-1, FunctionName: PlatformPublishingTwitterPublisher)"
time="2022-05-20T12:03:42+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Duration (region: eu-west-1, FunctionName: PlatformPublishingGooglePublisher)"
time="2022-05-20T12:03:42+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Duration (region: eu-west-1, FunctionName: PlatformPublishingCalendarEventIndexUpdater)"
time="2022-05-20T12:03:43+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Invocations (region: eu-west-1, FunctionName: PlatformPublishingVideoConversionStatusUpdater)"
time="2022-05-20T12:03:43+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Duration (region: eu-west-1, FunctionName: PlatformPublishingYoutubePublisher)"
time="2022-05-20T12:03:43+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Duration (region: eu-west-1, FunctionName: PlatformPublishingPostIndexer)"
time="2022-05-20T12:03:43+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Duration (region: eu-west-1, FunctionName: PlatformPublishingQueueMigrator)"
time="2022-05-20T12:03:43+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Duration (region: eu-west-1, FunctionName: PlatformPublishingQueueItemMigrator)"
time="2022-05-20T12:03:43+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Duration (region: eu-west-1, FunctionName: PlatformPublishingInstagramPublisher)"
time="2022-05-20T12:03:43+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Duration (region: eu-west-1, FunctionName: PlatformPublishingUnapprovedQueuedPostSkipper)"
time="2022-05-20T12:03:43+02:00" level=debug msg="Querying AWS CloudWatch: AWS/Lambda Duration (region: eu-west-1, FunctionName: PlatformPublishingVideoConverter)"
panic: interface conversion: interface {} is []*schema.UsageItem, not *usage.ResourceUsage

goroutine 74 [running]:
github.com/infracost/infracost/internal/usage.replaceResourceUsages(0x14000c307e0, 0x14000ecdf10, {0x98?})
        github.com/infracost/infracost/internal/usage/sync.go:256 +0x618
github.com/infracost/infracost/internal/usage.syncResource(0x14000596000, 0x140007a0c60, 0x0?, 0x104a84594?)
        github.com/infracost/infracost/internal/usage/sync.go:196 +0x11c
github.com/infracost/infracost/internal/usage.syncResourceUsages.func1(0x0?, 0x14000125f90?)
        github.com/infracost/infracost/internal/usage/sync.go:116 +0x5c
created by github.com/infracost/infracost/internal/usage.syncResourceUsages
        github.com/infracost/infracost/internal/usage/sync.go:114 +0x3b0
```

**Comment by vdmgolub** [comment]
@maathor Hello! Thank you for creating the issue! This error doesn't look right.

I've tried to run Infracost with `--sync-usage-file` on our infra project and it worked okay. I wonder if it's possible to provide Terraform code related to this run so we could reproduce the error on our end? It failed after `PlatformPublishingVideoConverter` log line, so maybe there is something specific about it? Any help would be appreciated.

**Comment by vdmgolub** [comment]
@maathor Thank you for providing the code examples! Knowing that you're using the Elastic Beanstalk Environment (I missed its mention in the log line above 😅) I was able to locate the issue:

The usage schema is expected to be `*usage.ResourceUsage`, but it actually is a slice of usage items instead: https://github.com/infracost/infracost/blob/master/internal/resources/aws/elastic_beanstalk_environment.go#L32
The fix would be wrapping them in
```
DefaultValue: &usage.ResourceUsage{Name: "cloudwatch", Items: schema.SubResourceUsage}CloudwatchLogGroupUsageSchema},
```
I haven't replicated the error yet, but seems like it's the reason.

We'll fix that, and it should be out with the next release. Thank you for all your help! :) We'll notify here when it is out.

**Comment by hugorut** [comment]
@maathor The fix for this is live in the [v0.10.1](https://github.com/infracost/infracost/releases/tag/v0.10.1) release.
